File: app/src/main/java/com/frogobox/cleaner/view/adapter/PowerViewAdapter.java
package com.frogobox.cleaner.view.adapter;

import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.frogobox.cleaner.model.PowerItem;

import com.frogobox.cleaner.myapplication.R;

import java.util.List;

/**
 * Created by Frogobox Software Industries 2/16/2017.
 */

public class PowerViewAdapter extends RecyclerView.Adapter<PowerViewAdapter.MyViewHolder> {

    public List<PowerItem> apps;

    public PowerViewAdapter(List<PowerItem> getapps)
    {
        apps=getapps;
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.powe_itemlist, parent, false);
        return new MyViewHolder(itemView);
    }



    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {

        PowerItem app= apps.get(position);
        holder.size.setText(app.getText());
    }

    @Override
    public int getItemCount() {
        return apps.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        public TextView size;


        public MyViewHolder(View view) {
            super(view);
            size = (TextView) view.findViewById(R.id.items);
        }
    }
}

File: app/src/main/java/com/frogobox/cleaner/view/adapter/PowerViewAdapter.kt
package com.frogobox.cleaner.view.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.frogobox.cleaner.R
import com.frogobox.cleaner.model.PowerItem
import kotlinx.android.synthetic.main.content_item_power.view.*

/**
 * Created by Frogobox Software Industries 2/16/2017.
 */

class PowerViewAdapter(private val apps: List<PowerItem>) :
        RecyclerView.Adapter<PowerViewAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.content_item_power, parent, false)
        return MyViewHolder(itemView)
    }


    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val (text) = apps[position]
        holder.size.text = text
    }

    override fun getItemCount(): Int {
        return apps.size
    }

    inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val size = view.items
    }

}
